index.js

```diff
@@ -161,36 +198,42 @@ S3Storage.prototype._handleFile = function (req, file, cb) {
       Metadata: opts.metadata,
       StorageClass: opts.storageClass,
       ServerSideEncryption: opts.serverSideEncryption,
-      SSEKMSKeyId: opts.sseKmsKeyId,
-      Body: (opts.replacementStream || file.stream)
+      SSEKMSKeyId: opts.sseKmsKeyId
     }
 
     if (opts.contentDisposition) {
       params.ContentDisposition = opts.contentDisposition
     }
 
-    var upload = this.s3.upload(params)
-
-    upload.on('httpUploadProgress', function (ev) {
-      if (ev.total) currentSize = ev.total
-    })
+    if (file.mimetype !== 'image/svg+xml') {
+      params.Body = stream
+      var upload = this.s3.upload(params)
+      return initializeUploadEvents(upload, opts, cb)
+    }
 
-    upload.send(function (err, result) {
-      if (err) return cb(err)
+    var streamAsString = ''
+    var fileSize = 0
+    const MAX_FILE_SIZE = 1000000000000
 
-      cb(null, {
-        size: currentSize,
-        bucket: opts.bucket,
-        key: opts.key,
-        acl: opts.acl,
-        contentType: opts.contentType,
-        contentDisposition: opts.contentDisposition,
-        storageClass: opts.storageClass,
-        serverSideEncryption: opts.serverSideEncryption,
-        metadata: opts.metadata,
-        location: result.Location,
-        etag: result.ETag
-      })
+    stream.on('data', data => {
+      fileSize += data.length
+      if (fileSize > MAX_FILE_SIZE) {
+        stream.emit('error', new Error('Filesize exceeded limit'))
+      }
+      streamAsString += data
+    })
+    stream.on('error', err => {
+      cb(err)
+    })
+    stream.on('close', () => {
```

Comment:
is it possible to have both an `error` event and a `close` or `end` event?

I don't actually know.